src/templates/ar/2019/base.html

```diff
@@ -97,10 +146,45 @@
         var menuIcon = document.querySelector('.menu-icon');
         var menuIconInitialSrc = document.querySelector('.menu-icon').src;
         var menuIconSrcClose = document.querySelector('.menu-icon').dataset.srcClose;
+        var menuBtn = document.querySelector('.menu-btn');
+        var menuNav = document.querySelector('#Menu');
+        var menu = document.querySelector('#Menu');
+
+        function triggerMenu() {
+          menuBtn.click();
+        }
 
-        document.querySelector('.menu-btn').addEventListener('click', function() {
+        menuBtn.addEventListener('click', function() {
           var menuOpen = document.body.classList.toggle('menu-open');
           menuIcon.src = menuOpen ? menuIconSrcClose : menuIconInitialSrc;
+          menuBtn.setAttribute('aria-expanded',menuOpen);
+          var ariaLabel = menuOpen ?  "Close the menu" : "Open the menu";
+          menuBtn.setAttribute('aria-label',ariaLabel);
+          event.stopPropagation();
+
+          /* When you open the menu, add an event listener to close it when clicking outside the menu area */
+          /* Remove it on closing the menu */
+          if (menuBtn.getAttribute('aria-expanded') === 'true') {
+            document.body.addEventListener('click', triggerMenu, false);
+          } else {
+            document.body.removeEventListener('click', triggerMenu, false);
+          }
+
+        });
+        
+        /* Add a click listener to menu so when it's open it swallows click to avoid above click closing it */
+        menu.addEventListener("click", function () {
+          event.stopPropagation();
+        });
+        
+        
+        menuNav.addEventListener("keydown", function onPress(event) {
```

Comment:
Any reason not to make this function anonymous?

```suggestion
        menuNav.addEventListener('keydown', function (event) {
```

Reply:
None at all!

Fixed in https://github.com/HTTPArchive/almanac.httparchive.org/pull/666/commits/1d5172131674bfb0a4e5f21ed6431d4a09e56eef